src/features/roles/roles.js

```diff
@@ -184,6 +189,16 @@ const Roles = () => {
   // used for (not) reseting the filters after submit
   const removingAllRows = pagination.count === removeRolesList.length;
 
+  // Show NotAuthorized component for users without proper permissions
+  if (!orgAdmin && !userAccessAdministrator) {
```

Comment:
I am not sure about the `orgAdmin` I think it should be better to check for org admin or user access admin
```suggestion
  if (!orgAdmin || !userAccessAdministrator) {
```